packages/nextjs/components/scaffold-eth/Contract/MethodSelector.tsx

```diff
@@ -0,0 +1,124 @@
+import { useState } from "react";
+import { Abi } from "viem";
+import { ChevronDownIcon, ChevronRightIcon, XMarkIcon } from "@heroicons/react/24/outline";
+
+type AbiFunction = Extract<Abi[number], { type: "function" }>;
+
+type AbiFallback = Extract<Abi[number], { type: "fallback" }>;
+
+type SelectableAbiType = AbiFunction | AbiFallback;
+
+interface MethodSelectorProps {
+  methodsWithInputs: SelectableAbiType[];
```

Comment:
Incase we go with https://github.com/BuidlGuidl/abi.ninja/pull/36/files#r1466811149 we should change it to `AbiFunction[]`